test/sodarouter-test.js

```diff
@@ -0,0 +1,129 @@
+'use strict';
+require('./mock-env');
+
+const expect = require('chai').expect;
+const superagent = require('superagent');
+const Soda = require('../model/soda.js');
+require('../server');
+const baseURL = `http://localhost:${process.env.PORT}`;
+
+//*****************POST TESTING************************
+describe('Testing Post route', function(){
+  afterEach((done) => {
+    Soda.remove({})
+    .then( () => done())
+    .catch(done);
+  });
+  describe('testing /api/soda', function(){
+
+    it('should return a soda with valid body',(done) => {
+      superagent.post(`${baseURL}/api/soda`)
+      .send({
+        brand: 'RC',
+        calories: 230,
+        diet: false,
+        taste: 'too sweet',
+      })
+    .then(res => {
+      expect(res.status).to.equal(200);
+      expect(res.body.brand).to.equal('RC');
+      expect(res.body.calories).to.equal(230);
+      expect(res.body.diet).to.equal(false);
+      expect(res.body.taste).to.equal('too sweet');
+      done();
+    })
+    .catch(done);
+    }); //end of it block
+    it('should return a 400 status',(done) => {
+      superagent.post(`${baseURL}/api/soda`)
+      .send({
+        brand: 'RC',
+      })
+    .then(done)
+    .catch(err => {
+      expect(err.status).to.equal(400);
+      done();
+    })
+    .catch(done);
+    }); //end of it block
+  });
+});
+
+//*************GET TEST**************************
+describe('testing GET Route', function(){
+  beforeEach((done) => {
+    new Soda ({
+      brand: 'Diet Pepsi',
+      calories: 1,
+      diet: true,
+      taste: 'ewwwww',
+    }).save()
+    .then(soda => {
+      this.fakeSoda = soda;
+      done();
+    })
+    .catch(done);
```

Comment:
Since you've separated out your routes in their own describe blocks, remember to clean up your test data for each block so you have an empty database after every test. Right now, only your POST tests clean up the data, so after every ``beforeEach`` in the other describe blocks, you would also use an ``afterEach`` block.

```
afterEach((done) => {
    Soda.remove({})
    .then( () => done())
    .catch(done);
  });
```